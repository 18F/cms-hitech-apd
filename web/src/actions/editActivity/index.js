export * from './contractorResources';
export * from './costAllocate';
export * from './overview';
