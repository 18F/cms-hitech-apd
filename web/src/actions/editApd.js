import { updateBudget } from './apd';

export const ADD_APD_ITEM = Symbol('add apd item');
export const EDIT_APD = Symbol('edit apd');
export const REMOVE_APD_ITEM = Symbol('remove apd item');

export const addKeyPerson = () => (dispatch, getState) => {
  dispatch({
    type: ADD_APD_ITEM,
    path: '/keyPersonnel/-',
    state: getState()
  });
};

export const removeKeyPerson = (
  index,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_ITEM,
      path: `/keyPersonnel/${index}`
    });
    dispatch(updateBudget());
  }
};

export const setKeyPersonName = (name, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/name`,
  value: name
});

export const setKeyPersonEmail = (email, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/email`,
  value: email
});

export const setKeyPersonRole = (role, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/position`,
  value: role
});

export const setKeyPersonPercentTime = (percent, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/percentTime`,
  value: percent
});

// needs to trigger budget calculation
export const setKeyPersonHasCosts = (hasCosts, index) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/hasCosts`,
    value: hasCosts
  });
  dispatch(updateBudget());
};

// needs to trigger budget calculation
export const setKeyPersonCost = (cost, index, year) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/costs/${year}`,
    value: cost
  });
  dispatch(updateBudget());
};

export const setMedicaidDirectorEmail = email => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/email',
    value: email
  };
};

export const setMedicaidDirectorName = name => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/name',
    value: name
  };
};

export const setMedicaidDirectorPhoneNumber = phone => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/phone',
    value: phone
  };
};

export const setMedicaidOfficeAddress1 = address => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/address1',
    value: address
  };
};

export const setMedicaidOfficeAddress2 = address => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/address2',
    value: address
  };
};

export const setMedicaidOfficeCity = city => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/city',
    value: city
  };
};

export const setMedicaidOfficeState = state => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/state',
    value: state
  };
};

export const setMedicaidOfficeZip = zip => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/zip',
    value: zip
  };
};
