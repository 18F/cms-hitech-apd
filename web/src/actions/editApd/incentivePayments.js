import { EDIT_APD } from './symbols';

export const setIncentiveEHCount = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/ehCt/${year}/${quarter}`,
  value
});

export const setIncentiveEHPayment = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/ehAmt/${year}/${quarter}`,
  value
});

export const setIncentiveEPCount = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/epCt/${year}/${quarter}`,
  value
});

export const setIncentiveEPPayment = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/epAmt/${year}/${quarter}`,
  value
});
