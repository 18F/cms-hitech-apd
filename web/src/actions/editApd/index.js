export * from './symbols';

export * from './keyPersonnel';
export * from './overview';
export * from './previousActivities';
export * from './stateProfile';
