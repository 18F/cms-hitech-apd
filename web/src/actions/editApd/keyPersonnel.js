import { ADD_APD_ITEM, EDIT_APD, REMOVE_APD_ITEM } from './symbols';
import { updateBudget } from '../apd';

export const addKeyPerson = () => (dispatch, getState) => {
  dispatch({
    type: ADD_APD_ITEM,
    path: '/keyPersonnel/-',
    state: getState()
  });
};

export const removeKeyPerson = (
  index,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_ITEM,
      path: `/keyPersonnel/${index}`
    });
    dispatch(updateBudget());
  }
};

export const setKeyPersonName = (name, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/name`,
  value: name
});

export const setKeyPersonEmail = (email, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/email`,
  value: email
});

export const setKeyPersonRole = (role, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/position`,
  value: role
});

export const setKeyPersonPercentTime = (percent, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/percentTime`,
  value: percent
});

// needs to trigger budget calculation
export const setKeyPersonHasCosts = (hasCosts, index) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/hasCosts`,
    value: hasCosts
  });
  dispatch(updateBudget());
};

// needs to trigger budget calculation
export const setKeyPersonCost = (cost, index, year) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/costs/${year}`,
    value: cost
  });
  dispatch(updateBudget());
};
