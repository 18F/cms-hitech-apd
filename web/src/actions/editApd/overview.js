import { ADD_APD_YEAR, EDIT_APD, REMOVE_APD_YEAR } from './symbols';
import { updateBudget } from '../apd';

export const addYear = year => (dispatch, getState) => {
  dispatch({ type: ADD_APD_YEAR, value: year, state: getState() });
  dispatch(updateBudget());
};

export const removeYear = year => (dispatch, getState) => {
  dispatch({ type: REMOVE_APD_YEAR, value: year, state: getState() });
  dispatch(updateBudget());
};

export const setNarrativeForHIE = text => ({
  type: EDIT_APD,
  path: '/narrativeHIE',
  value: text
});

export const setNarrativeForHIT = text => ({
  type: EDIT_APD,
  path: '/narrativeHIT',
  value: text
});

export const setNarrativeForMMIS = text => ({
  type: EDIT_APD,
  path: '/narrativeMMIS',
  value: text
});

export const setProgramOverview = text => ({
  type: EDIT_APD,
  path: '/programOverview',
  value: text
});
