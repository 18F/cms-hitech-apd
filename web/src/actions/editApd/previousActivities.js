import { EDIT_APD } from './symbols';

export const setPreviousActivitySummary = summary => ({
  type: EDIT_APD,
  path: '/previousActivitySummary',
  value: summary
});

export const setPreviousActivityApprovedExpenseForHITandHIE = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/hithie/totalApproved`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseForHITandHIE = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/hithie/federalActual`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS50FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/50/totalApproved`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS75FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/75/totalApproved`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS90FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/90/totalApproved`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS50FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/50/federalActual`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS75FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/75/federalActual`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS90FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/90/federalActual`,
  value: expense
});
