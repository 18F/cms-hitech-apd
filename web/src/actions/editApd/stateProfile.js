import { EDIT_APD } from './symbols';

export const setMedicaidDirectorEmail = email => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidDirector/email',
  value: email
});

export const setMedicaidDirectorName = name => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidDirector/name',
  value: name
});

export const setMedicaidDirectorPhoneNumber = phone => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidDirector/phone',
  value: phone
});

export const setMedicaidOfficeAddress1 = address => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/address1',
  value: address
});

export const setMedicaidOfficeAddress2 = address => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/address2',
  value: address
});

export const setMedicaidOfficeCity = city => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/city',
  value: city
});

export const setMedicaidOfficeState = state => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/state',
  value: state
});

export const setMedicaidOfficeZip = zip => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/zip',
    value: zip
  };
};
