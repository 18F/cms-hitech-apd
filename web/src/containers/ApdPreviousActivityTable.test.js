import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import ApdPreviousActivityTable from './ApdPreviousActivityTable';
import { EDIT_APD } from '../actions/editApd';

const mockStore = configureStore([thunk]);

describe('apd previous activity table, mmis component', () => {
  const state = {
    apd: {
      data: {
        previousActivityExpenses: {
          '1': {
            hithie: {
              federalActual: 10,
              totalApproved: 20
            }
          },
          '2': {
            hithie: {
              federalActual: 100,
              totalApproved: 200
            }
          }
        }
      }
    }
  };

  const store = mockStore(state);

  beforeEach(() => {
    store.clearActions();
  });

  test('renders correctly', () => {
    expect(
      mount(
        <Provider store={store}>
          <ApdPreviousActivityTable />
        </Provider>
      )
    ).toMatchSnapshot();
  });

  test('dispatches on a change', () => {
    mount(
      <Provider store={store}>
        <ApdPreviousActivityTable />
      </Provider>
    )
      .find('DollarField[name="hithie-approved-total-1"]')
      .prop('onChange')({ target: { value: 'new value' } });

    expect(store.getActions()).toEqual([
      {
        type: EDIT_APD,
        path: '/previousActivityExpenses/1/hithie/totalApproved',
        value: 'new value'
      }
    ]);
  });
});
