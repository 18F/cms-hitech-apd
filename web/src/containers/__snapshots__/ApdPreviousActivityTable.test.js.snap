// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apd previous activity table, mmis component renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ApdPreviousActivityTable>
    <table
      className="budget-table"
    >
      <caption
        className="ds-h4"
      >
        HIT + HIE
      </caption>
      <thead>
        <tr>
          <th
            id="prev_act_hit_header_ffy"
          >
            <span
              className="ds-u-visibility--screen-reader"
            >
              Year
            </span>
          </th>
          <th
            id="prev_act_hithie_total"
          >
            <span>
              Total computable
              <br />
              (Federal + state)
            </span>
          </th>
          <th
            colSpan="2"
            id="prev_act_hithie_federal"
          >
            <span>
              Federal share 
              <strong>
                90
                %
              </strong>
               FFP
            </span>
          </th>
        </tr>
        <tr>
          <th
            aria-hidden="true"
          />
          <th
            className="ds-u-text-align--right"
            id="prev_act_hithie_total_approved"
          >
            Approved
          </th>
          <th
            className="ds-u-text-align--right"
            id="prev_act_hithie_federal_approved"
          >
            Approved
          </th>
          <th
            className="ds-u-text-align--right"
            id="prev_act_hithie_federal_actual"
          >
            Actual
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="1"
        >
          <th
            headers="prev_act_hit_header_ffy"
            id="prev_act_hithie_row_1"
          >
            FFY 1
          </th>
          <td
            headers="prev_act_hithie_row_1 prev_act_hithie_total prev_act_hithie_total_approved"
          >
            <DollarField
              className="budget-table--input-holder"
              fieldClassName="budget-table--input__number"
              label="total approved funding for HIT and HIE for FFY 1, state plus federal"
              labelClassName="sr-only"
              name="hithie-approved-total-1"
              onBlur={null}
              onChange={[Function]}
              value={20}
            >
              <NumberField
                className="budget-table--input-holder"
                fieldClassName="budget-table--input__number"
                fieldRef={null}
                label="total approved funding for HIT and HIE for FFY 1, state plus federal"
                labelClassName="sr-only"
                mask="currency"
                name="hithie-approved-total-1"
                onBlur={[Function]}
                onChange={[Function]}
                value="20"
              >
                <TextField
                  className="budget-table--input-holder"
                  fieldClassName="budget-table--input__number"
                  fieldRef={
                    Object {
                      "current": <input
                        aria-label="total approved funding for HIT and HIE for FFY 1, state plus federal. Enter amount in dollars."
                        class="ds-c-field ds-c-field--currency budget-table--input__number"
                        id="textfield_1"
                        name="hithie-approved-total-1"
                        type="text"
                        value="20"
                      />,
                    }
                  }
                  label="total approved funding for HIT and HIE for FFY 1, state plus federal"
                  labelClassName="sr-only"
                  mask="currency"
                  name="hithie-approved-total-1"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="20"
                >
                  <div
                    className="ds-u-clearfix budget-table--input-holder"
                  >
                    <FormLabel
                      className="sr-only"
                      component="label"
                      fieldId="textfield_1"
                    >
                      <label
                        className="ds-c-label sr-only"
                        htmlFor="textfield_1"
                      >
                        <span
                          className=""
                        >
                          total approved funding for HIT and HIE for FFY 1, state plus federal
                        </span>
                      </label>
                    </FormLabel>
                    <div
                      className="ds-c-field-mask ds-c-field-mask--currency"
                    >
                      <div
                        className="ds-c-field__before ds-c-field__before--currency"
                      >
                        $
                      </div>
                      <Mask
                        mask="currency"
                      >
                        <input
                          aria-label="total approved funding for HIT and HIE for FFY 1, state plus federal. Enter amount in dollars."
                          className="ds-c-field ds-c-field--currency budget-table--input__number"
                          id="textfield_1"
                          name="hithie-approved-total-1"
                          onBlur={[Function]}
                          onChange={[Function]}
                          type="text"
                          value="20"
                        />
                      </Mask>
                    </div>
                  </div>
                </TextField>
              </NumberField>
            </DollarField>
          </td>
          <td
            className="budget-table--number"
            headers="prev_act_hithie_row_1 prev_act_hithie_federal prev_act_hithie_federal_approved"
          >
            <Dollars
              long={false}
            >
              <span
                className="visibility--screen"
              >
                $18
              </span>
              <span
                className="visibility--print"
              >
                $18
              </span>
            </Dollars>
          </td>
          <td
            headers="prev_act_hithie_row_1 prev_act_hithie_federal prev_act_hithie_federal_actual"
          >
            <DollarField
              className="budget-table--input-holder"
              fieldClassName="budget-table--input__number"
              label="actual federal share for HIT and HIE for FFY 1"
              labelClassName="sr-only"
              name="hithie-actual-federal-1"
              onBlur={null}
              onChange={[Function]}
              value={10}
            >
              <NumberField
                className="budget-table--input-holder"
                fieldClassName="budget-table--input__number"
                fieldRef={null}
                label="actual federal share for HIT and HIE for FFY 1"
                labelClassName="sr-only"
                mask="currency"
                name="hithie-actual-federal-1"
                onBlur={[Function]}
                onChange={[Function]}
                value="10"
              >
                <TextField
                  className="budget-table--input-holder"
                  fieldClassName="budget-table--input__number"
                  fieldRef={
                    Object {
                      "current": <input
                        aria-label="actual federal share for HIT and HIE for FFY 1. Enter amount in dollars."
                        class="ds-c-field ds-c-field--currency budget-table--input__number"
                        id="textfield_2"
                        name="hithie-actual-federal-1"
                        type="text"
                        value="10"
                      />,
                    }
                  }
                  label="actual federal share for HIT and HIE for FFY 1"
                  labelClassName="sr-only"
                  mask="currency"
                  name="hithie-actual-federal-1"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="10"
                >
                  <div
                    className="ds-u-clearfix budget-table--input-holder"
                  >
                    <FormLabel
                      className="sr-only"
                      component="label"
                      fieldId="textfield_2"
                    >
                      <label
                        className="ds-c-label sr-only"
                        htmlFor="textfield_2"
                      >
                        <span
                          className=""
                        >
                          actual federal share for HIT and HIE for FFY 1
                        </span>
                      </label>
                    </FormLabel>
                    <div
                      className="ds-c-field-mask ds-c-field-mask--currency"
                    >
                      <div
                        className="ds-c-field__before ds-c-field__before--currency"
                      >
                        $
                      </div>
                      <Mask
                        mask="currency"
                      >
                        <input
                          aria-label="actual federal share for HIT and HIE for FFY 1. Enter amount in dollars."
                          className="ds-c-field ds-c-field--currency budget-table--input__number"
                          id="textfield_2"
                          name="hithie-actual-federal-1"
                          onBlur={[Function]}
                          onChange={[Function]}
                          type="text"
                          value="10"
                        />
                      </Mask>
                    </div>
                  </div>
                </TextField>
              </NumberField>
            </DollarField>
          </td>
        </tr>
        <tr
          key="2"
        >
          <th
            headers="prev_act_hit_header_ffy"
            id="prev_act_hithie_row_2"
          >
            FFY 2
          </th>
          <td
            headers="prev_act_hithie_row_2 prev_act_hithie_total prev_act_hithie_total_approved"
          >
            <DollarField
              className="budget-table--input-holder"
              fieldClassName="budget-table--input__number"
              label="total approved funding for HIT and HIE for FFY 2, state plus federal"
              labelClassName="sr-only"
              name="hithie-approved-total-2"
              onBlur={null}
              onChange={[Function]}
              value={200}
            >
              <NumberField
                className="budget-table--input-holder"
                fieldClassName="budget-table--input__number"
                fieldRef={null}
                label="total approved funding for HIT and HIE for FFY 2, state plus federal"
                labelClassName="sr-only"
                mask="currency"
                name="hithie-approved-total-2"
                onBlur={[Function]}
                onChange={[Function]}
                value="200"
              >
                <TextField
                  className="budget-table--input-holder"
                  fieldClassName="budget-table--input__number"
                  fieldRef={
                    Object {
                      "current": <input
                        aria-label="total approved funding for HIT and HIE for FFY 2, state plus federal. Enter amount in dollars."
                        class="ds-c-field ds-c-field--currency budget-table--input__number"
                        id="textfield_3"
                        name="hithie-approved-total-2"
                        type="text"
                        value="200"
                      />,
                    }
                  }
                  label="total approved funding for HIT and HIE for FFY 2, state plus federal"
                  labelClassName="sr-only"
                  mask="currency"
                  name="hithie-approved-total-2"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="200"
                >
                  <div
                    className="ds-u-clearfix budget-table--input-holder"
                  >
                    <FormLabel
                      className="sr-only"
                      component="label"
                      fieldId="textfield_3"
                    >
                      <label
                        className="ds-c-label sr-only"
                        htmlFor="textfield_3"
                      >
                        <span
                          className=""
                        >
                          total approved funding for HIT and HIE for FFY 2, state plus federal
                        </span>
                      </label>
                    </FormLabel>
                    <div
                      className="ds-c-field-mask ds-c-field-mask--currency"
                    >
                      <div
                        className="ds-c-field__before ds-c-field__before--currency"
                      >
                        $
                      </div>
                      <Mask
                        mask="currency"
                      >
                        <input
                          aria-label="total approved funding for HIT and HIE for FFY 2, state plus federal. Enter amount in dollars."
                          className="ds-c-field ds-c-field--currency budget-table--input__number"
                          id="textfield_3"
                          name="hithie-approved-total-2"
                          onBlur={[Function]}
                          onChange={[Function]}
                          type="text"
                          value="200"
                        />
                      </Mask>
                    </div>
                  </div>
                </TextField>
              </NumberField>
            </DollarField>
          </td>
          <td
            className="budget-table--number"
            headers="prev_act_hithie_row_2 prev_act_hithie_federal prev_act_hithie_federal_approved"
          >
            <Dollars
              long={false}
            >
              <span
                className="visibility--screen"
              >
                $180
              </span>
              <span
                className="visibility--print"
              >
                $180
              </span>
            </Dollars>
          </td>
          <td
            headers="prev_act_hithie_row_2 prev_act_hithie_federal prev_act_hithie_federal_actual"
          >
            <DollarField
              className="budget-table--input-holder"
              fieldClassName="budget-table--input__number"
              label="actual federal share for HIT and HIE for FFY 2"
              labelClassName="sr-only"
              name="hithie-actual-federal-2"
              onBlur={null}
              onChange={[Function]}
              value={100}
            >
              <NumberField
                className="budget-table--input-holder"
                fieldClassName="budget-table--input__number"
                fieldRef={null}
                label="actual federal share for HIT and HIE for FFY 2"
                labelClassName="sr-only"
                mask="currency"
                name="hithie-actual-federal-2"
                onBlur={[Function]}
                onChange={[Function]}
                value="100"
              >
                <TextField
                  className="budget-table--input-holder"
                  fieldClassName="budget-table--input__number"
                  fieldRef={
                    Object {
                      "current": <input
                        aria-label="actual federal share for HIT and HIE for FFY 2. Enter amount in dollars."
                        class="ds-c-field ds-c-field--currency budget-table--input__number"
                        id="textfield_4"
                        name="hithie-actual-federal-2"
                        type="text"
                        value="100"
                      />,
                    }
                  }
                  label="actual federal share for HIT and HIE for FFY 2"
                  labelClassName="sr-only"
                  mask="currency"
                  name="hithie-actual-federal-2"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="100"
                >
                  <div
                    className="ds-u-clearfix budget-table--input-holder"
                  >
                    <FormLabel
                      className="sr-only"
                      component="label"
                      fieldId="textfield_4"
                    >
                      <label
                        className="ds-c-label sr-only"
                        htmlFor="textfield_4"
                      >
                        <span
                          className=""
                        >
                          actual federal share for HIT and HIE for FFY 2
                        </span>
                      </label>
                    </FormLabel>
                    <div
                      className="ds-c-field-mask ds-c-field-mask--currency"
                    >
                      <div
                        className="ds-c-field__before ds-c-field__before--currency"
                      >
                        $
                      </div>
                      <Mask
                        mask="currency"
                      >
                        <input
                          aria-label="actual federal share for HIT and HIE for FFY 2. Enter amount in dollars."
                          className="ds-c-field ds-c-field--currency budget-table--input__number"
                          id="textfield_4"
                          name="hithie-actual-federal-2"
                          onBlur={[Function]}
                          onChange={[Function]}
                          type="text"
                          value="100"
                        />
                      </Mask>
                    </div>
                  </div>
                </TextField>
              </NumberField>
            </DollarField>
          </td>
        </tr>
      </tbody>
    </table>
  </ApdPreviousActivityTable>
</Provider>
`;
